cranelift/codegen/src/machinst/lower.rs

```diff
@@ -142,13 +143,16 @@ pub trait LowerBackend {
 
 /// Machine-independent lowering driver / machine-instruction container. Maintains a correspondence
 /// from original Inst to MachInsts.
-pub struct Lower<'func, I: VCodeInst> {
+pub struct Lower<'func, 'env, I: VCodeInst> {
     /// The function to lower.
     f: &'func Function,
 
     /// Machine-independent flags.
     flags: crate::settings::Flags,
 
+    /// The MachineEnv used when allocating registers.
+    machine_env: &'env MachineEnv,
+
```

Comment:
Did you try reusing `'func` here by chance and find that a new lifetime was actually necessary to get the code to compile? This is pretty invasive and adds a third(!!) lifetime to the `IsleContext` so if it isn't actually necessary it would be nice to skip.

Reply:
That worked, thanks for pointing this out!

Reply:
Now that I'm thinking about this, we don't need the actual `MachineEnv` plumbed through, just the `PRegSet` that's used in the `OperandCollector`. Moving the creation of the `PRegSet` to `Lower::new` removes the `MachineEnv` reference, and only iterates it once instead of for each instruction.